### PR TITLE
fix(TDI-41573): Enable secure processing + disallow doctypes for DocumentBuilderFactory instances

### PR DESCRIPTION
**Pull request by wwang-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-41573

**What is the new behavior?**
https://jira.talendforge.org/browse/TDI-41573

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/pom.xml [hunk 1]
```diff
@@ -55,6 +55,11 @@
     		<artifactId>httpclient</artifactId>
     		<version>4.5.2</version>
 		</dependency>
+		<dependency>
```
Comment (coheigea) [inline main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/pom.xml]: Why import httpcore here? An older version comes in from httpclient 4.5.2 above. If necessary we could upgrade httpclient instead?

File: main/plugins/org.talend.designer.components.localprovider/components/tDTDValidator/tDTDValidator_main.javajet [hunk 2]
```diff
@@ -64,6 +64,13 @@ String xmlfile = ElementParameterParser.getValue(node, "__XMLFILE__");
 		}
 		
 		dbf<%=cid %> = javax.xml.parsers.DocumentBuilderFactory.newInstance();
+		
+		try {
+			dbf<%=cid %>.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+		} catch (java.lang.Exception e_<%=cid%>) {
```
Comment (coheigea) [inline main/plugins/org.talend.designer.components.localprovider/components/tDTDValidator/tDTDValidator_main.javajet]: I don't think we should set the secure processing feature here.

File: main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/pom.xml (hunk 1 above)
Comment (wwang-talend) [reply main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/pom.xml]: strange, i run the mvn clean install at my local, miss lots of httpcore classes, yes, i think httpclient should introduce httpore, but not in fact.

Comment (wwang-talend) [reply main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/pom.xml]: here, httpcore version is only for compiler, but anyway, will adjust it to the right version

File: main/plugins/org.talend.designer.components.localprovider/components/tDTDValidator/tDTDValidator_main.javajet (hunk 2 above)
Comment (coheigea) [reply main/plugins/org.talend.designer.components.localprovider/components/tDTDValidator/tDTDValidator_main.javajet]: @wwang-talend - See my comment above, I think we don't need to add secure processing for this file.

File: main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/org/talend/ms/crm/sdk/WsdlTokenManager.java [hunk 3]
```diff
@@ -214,15 +213,24 @@ private String getValueFromXML(String inputXML, String xPathQuery)
...
         DocumentBuilder builder;
         try {
-            builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            
+            try {
+	            factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+	            factory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+            } catch (Exception e) {
+            	logger.warn("failed to ebable xml safe feature");
```
Comment (coheigea) [inline main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/org/talend/ms/crm/sdk/WsdlTokenManager.java]: "enable" not "ebable". Same mistake below.

File: main/plugins/org.talend.designer.components.libs/libs_src/soap/src/main/java/org/talend/soap/SOAPUtil.java [hunk 4]
```diff
@@ -208,6 +209,14 @@ public String getReBodyMessage() {
     private Document extractContentAsDocument(SOAPHeader header) throws ParserConfigurationException, TransformerException {
         Document document;
         DocumentBuilderFactory factory = new com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl();
```
Comment (coheigea) [inline main/plugins/org.talend.designer.components.libs/libs_src/soap/src/main/java/org/talend/soap/SOAPUtil.java]: Can we change this to DocumentBuilderFactory.newInstance() and get rid of the com.sun reference?

Comment (wwang-talend) [reply main/plugins/org.talend.designer.components.libs/libs_src/soap/src/main/java/org/talend/soap/SOAPUtil.java]: done, please help review again

File: main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/org/talend/ms/crm/sdk/WsdlTokenManager.java (hunk 3 above)
Comment (wwang-talend) [reply main/plugins/org.talend.designer.components.libs/libs_src/talend-mscrm/src/main/java/org/talend/ms/crm/sdk/WsdlTokenManager.java]: done, and create a jira for TUP https://jira.talendforge.org/browse/TUP-23403 for the grep above.

**Pull request merged**